File: tags/FUNNEL_005/libraries/ruby/funnel/xbee.rb
#!/usr/bin/env ruby -wKU

require "funnel/iosystem"

module Funnel
  class XBee < IOSystem
    (XBS1, XBS2) = Array(0..1)

    def self.XBS1
      return XBS1
    end

    def self.XBS2
      return XBS2
    end
    
    def initialize(nodes, host = 'localhost', port = 9000, interval = 33)
      super(nil, host, port, interval)
      @config = Configuration.new(Configuration::XBEE, XBS1)
      nodes = [] if nodes == nil
      nodes.each do |id|
        register_node(id, "")
      end
    end
    
    def register_node(id, ni)
      add_module(id, @config, ni)
    end
  end
end

File: tags/funnel_008b2/libraries/processing/src/processing/funnel/XBee.java
package processing.funnel;

import processing.core.PApplet;

public final class XBee extends IOSystem{

	
	public static final String moduleName = "Xbee";
	
	/**
	 * 0xFFFF : broadcast
	 */
	public static final int moduleID = 0xFFFF;//
	
	private static final int[] xbs1 = {
    PORT_AIN, PORT_AIN, PORT_AIN, PORT_AIN, PORT_AIN, PORT_AIN,
    PORT_DIN, PORT_DIN,
	};
	public static final Configuration XBS1 = new Configuration(moduleID,xbs1,moduleName);
	//ポートの機能(参照する名前)
	//private int analogInput[] = {0,1,2,3};
	//private int analogOutput[] = {10,11,12,13};


	public XBee(PApplet parent, String hostName,
			int commandPortNumber,int samplingInterval,int[] IDs,Configuration config){
		super(parent,hostName,commandPortNumber,samplingInterval,config);

		if(!initialize(moduleID,config)){
			errorMessage("Funnel configuration error!");
		}
		for(int i=0;i<IDs.length;i++){
			String name = "Fio." + i;
			addModule(IDs[i],config,name);
		}
		startIOSystem();
	}
	
	public XBee(PApplet parent,int[] IDs){	
		this(parent,"localhost",CommandPort.defaultPort,
				33,IDs,XBS1);
	}
	
	public XBee(PApplet parent, int[] IDs, Configuration config){
		this(parent,"localhost",CommandPort.defaultPort,
				33,IDs,config);
	}
	
	public XBee(PApplet parent, int samplingInterval,int[] IDs, Configuration config ){
		
		this(parent,"localhost",CommandPort.defaultPort,
				samplingInterval,IDs,config);
	}

	public XBee(PApplet parent,
			int commandPortNumber, int notifyPortNumber,int samplingInterval,int[] IDs, Configuration config ){
		
		this(parent,"localhost",commandPortNumber,
				samplingInterval,IDs,config);
	}

}
